### PR TITLE
[NeoMathEngine] remove 'using namespace' from headers

### PR DESCRIPTION
**Pull request by favorart**

### CODE REVIEW

File: NeoMathEngine/include/NeoMathEngine/NeoMathEngineDefs.h [hunk 1]
```diff
@@ -49,6 +49,7 @@ limitations under the License.
 
 #include <climits>
 #include <cstddef>
+#include <algorithm>
```
Comment (FedyuninV) [inline NeoMathEngine/include/NeoMathEngine/NeoMathEngineDefs.h]: This header belongs to external API, better add it to another place.
P.S. I'm not sure about __any__ includes here...

File: NeoMathEngine/src/CPU/x86/avx/src/xbyak/xbyak.h [hunk 2]
```diff
@@ -75,6 +75,9 @@
 	#ifndef WIN32_LEAN_AND_MEAN
 		#define WIN32_LEAN_AND_MEAN
 	#endif
+	#ifndef NOMINMAX
```
Comment (FedyuninV) [inline NeoMathEngine/src/CPU/x86/avx/src/xbyak/xbyak.h]: This is a third-party code. Are you __absolutely__ sure that you need to change it?

File: NeoMathEngine/src/CPU/x86/avx/src/BlobConvolution.inl [hunk 3]
```diff
@@ -338,20 +338,19 @@ std::vector<int> CBlobConvolution<FltCnt>::getPixelOffsetSrcSteps( int srcDim, i
...
+    return std::vector<int>( start, end );
 }
 
 template<int FltCnt>
 void CBlobConvolution<FltCnt>::fillPixelOffset()
 {
-    using namespace std;
-    vector<int> pixelOffsetSrcStepsX = getPixelOffsetSrcSteps( SrcW, FltW, DilationW, StrideW, PaddingW );
-    vector<int> pixelOffsetSrcStepsY = getPixelOffsetSrcSteps( SrcH, FltH, DilationH, StrideH, PaddingH );
+    std::vector<int> pixelOffsetSrcStepsX = getPixelOffsetSrcSteps( SrcW, FltW, DilationW, StrideW, PaddingW );
```
Comment (FedyuninV) [inline NeoMathEngine/src/CPU/x86/avx/src/BlobConvolution.inl:347]: Why? It is inside of a function...

Comment (favorart) [reply NeoMathEngine/src/CPU/x86/avx/src/BlobConvolution.inl:347]: For uniformity!
I have fixed everywhere, and also here. This is only function with that.
I know, that it limits the function area, however this is a 'using namaspace' in a header file. Not to stimulate use such constructions...
Should I revert this change?

Comment (FedyuninV) [reply NeoMathEngine/src/CPU/x86/avx/src/BlobConvolution.inl:347]: Not really. But other 2 comments are improtant.

File: NeoML/src/common.h [hunk 4]
```diff
@@ -18,5 +18,8 @@ limitations under the License.
 #include <NeoMathEngine/Platforms.h>
 
 #ifdef NEOML_USE_FINEOBJ
+#ifndef NOMINMAX
```
Comment (FedyuninV) [inline NeoML/src/common.h:20]: Why this is inside `NEOML_USE_FINEOBJ`?
Isn't it better to wrap it into
`#if FINE_PLATFORM(FINE_WINDOWS) && !defined(NOMINMAX)` outside of `#ifdef NEOML_USE_FINEOBJ`?

File: NeoMathEngine/include/NeoMathEngine/NeoMathEngineDefs.h (hunk 1 above)
Comment (favorart) [reply NeoMathEngine/include/NeoMathEngine/NeoMathEngineDefs.h]: additional include removed.

File: NeoMathEngine/src/CPU/x86/avx/src/xbyak/xbyak.h (hunk 2 above)
Comment (favorart) [reply NeoMathEngine/src/CPU/x86/avx/src/xbyak/xbyak.h]: NOMINMAX define replaced from library to upper level include file

File: NeoML/src/common.h (hunk 4 above)
Comment (favorart) [reply NeoML/src/common.h:20]: Fixed.
Put the NOMINMAX outside any other defines

**Review by SAngeliuk — APPROVED** [review]

**Review by FedyuninV — APPROVED** [review]

**Pull request merged**